intermediate_source/speech_command_recognition_with_torchaudio.py

```diff
@@ -0,0 +1,523 @@
+from pydub import AudioSegment
+from torchaudio.datasets import SPEECHCOMMANDS
+
+######################################################################
+# Let’s check if a CUDA GPU is available and select our device. Running
+# the network on a GPU will greatly decrease the training/testing runtime.
+#
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print(device)
+
+
+######################################################################
+# Importing the Dataset
+# ---------------------
+#
+# We use torchaudio to download and represent the dataset. Here we use
+# SpeechCommands, which is a datasets of 35 commands spoken by different
+# people. The dataset ``SPEECHCOMMANDS`` is a ``torch.utils.data.Dataset``
+# version of the dataset.
+#
+# The actual loading and formatting steps happen in the access function
+# ``__getitem__``. In ``__getitem__``, we use ``torchaudio.load()`` to
+# convert the audio files to tensors. ``torchaudio.load()`` returns a
+# tuple containing the newly created tensor along with the sampling
+# frequency of the audio file (16kHz for SpeechCommands). In this dataset,
+# all audio files are about 1 second long (and so about 16000 time frames
+# long).
+#
+# Here we wrap it to split it into standard training, validation, testing
+# subsets.
+#
+
+
+class SubsetSC(SPEECHCOMMANDS):
+    def __init__(self, subset: str = None):
+        super().__init__("./", download=True)
+
+        def load_list(filename):
+            filepath = os.path.join(self._path, filename)
+            with open(filepath) as fileobj:
+                return [os.path.join(self._path, line.strip()) for line in fileobj]
+
+        if subset == "validation":
+            self._walker = load_list("validation_list.txt")
+        elif subset == "testing":
+            self._walker = load_list("testing_list.txt")
+        elif subset == "training":
+            excludes = load_list("validation_list.txt") + load_list("testing_list.txt")
+            excludes = set(excludes)
+            self._walker = [w for w in self._walker if w not in excludes]
+
+
+# Create training and testing split of the data. We do not use validation in this tutorial.
+train_set = SubsetSC("training")
+test_set = SubsetSC("testing")
+
+waveform, sample_rate, label, speaker_id, utterance_number = train_set[0]
+
+
+######################################################################
+# A data point in the SPEECHCOMMANDS dataset is a tuple made of a waveform
+# (the audio signal), the sample rate, the utterance (label), the ID of
+# the speaker, the number of the utterance.
+#
+
+print("Shape of waveform: {}".format(waveform.size()))
+print("Sample rate of waveform: {}".format(sample_rate))
+
+plt.plot(waveform.t().numpy());
+
+
+######################################################################
+# Let’s find the list of labels available in the dataset.
+#
+
+labels = sorted(list(set(datapoint[2] for datapoint in train_set)))
```

Comment:
[not PR review comment] I wonder if the Dataset implementations should have this kind of attributes.

Reply:
(added mention in #910)